### PR TITLE
[Project] add DDTI dataset project in dev-1.x

### PR DESCRIPTION
**Pull request by Provable0816**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: projects/medical/2d_image/dermoscopy/ddti/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+```shell
+export PYTHONPATH=`pwd`:$PYTHONPATH
+```
+
+### Dataset Preparing
+
+- Download dataset from [here](https://drive.google.com/file/d/1wwlsEhwfSyvQsJBRjeDLhUjqZh8eaH2R/view) and save it to the `data/` directory .
+- Decompress data to path `data/`. This will create a new folder named `data/ddti/`, which contains the original image data.
+- run script `python tools/prepare_dataset.py` to format data and change folder structure as below.
+- run script `python ../../tools/split_seg_dataset.py` to split dataset. For the Bacteria_detection dataset, as there is no test or validation dataset, we sample 20% samples from the whole dataset as the validation dataset and 80% samples for training data and make two filename lists `train.txt` and `val.txt`. As we set the random seed as the hard code, we eliminated the randomness, the dataset split actually can be reproducible.
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/README.md:58]: Should update these steps.

File: projects/medical/2d_image/dermoscopy/ddti/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+### ddti
+
+***Note: The following experimental results are based on the data randomly partitioned according to the above method described in the dataset preparing section.***
+
+|     Method      | Backbone | Crop Size |   lr   | mIoU  | mDice |                                  config                                   |         download         |
+| :-------------: | :------: | :-------: | :----: | :---: | :---: | :-----------------------------------------------------------------------: | :----------------------: |
+| fcn_unet_s5-d16 |   unet   |  512x512  |  0.01  | 76.48 | 84.68 |  [config](./configs/fcn-unet-s5-d16_unet_1xb16-0.01-20k_ddti-512x512.py)  | [model](<>) \| [log](<>) |
+| fcn_unet_s5-d16 |   unet   |  512x512  | 0.001  | 61.06 | 63.69 | [config](./configs/fcn-unet-s5-d16_unet_1xb16-0.001-20k_ddti-512x512.py)  | [model](<>) \| [log](<>) |
+| fcn_unet_s5-d16 |   unet   |  512x512  | 0.0001 | 58.87 | 62.42 | [config](./configs/fcn-unet-s5-d16_unet_1xb16-0.0001-20k_ddti-512x512.py) | [model](<>) \| [log](<>) |
+
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/README.md:126]: Might remove these results.

File: projects/medical/2d_image/dermoscopy/ddti/tools/ltb_process.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    all_imgs = []
+    for fpath, dirname, fnames in os.walk(os.path.join(data_root, 'images')):
+        for fname in fnames:
+            all_imgs.append(os.path.join(fpath, fname))
+    f_ = open(data_root + '/images_md5_list.txt', 'w')
+    for img in sorted(all_imgs):
+        with open(img, 'rb') as fd:
+            fmd5 = hashlib.md5(fd.read()).hexdigest()
+        f_.write(fmd5 + '\t' + '/'.join(img.split('/')[-2:]) + '\n')
+    f_.close()
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/tools/ltb_process.py:54]: Will this part be deleted?

File: projects/medical/2d_image/dermoscopy/ddti/tools/prepare_dataset.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            imn[imn == num] = num_map[num]
+        new_im = Image.fromarray(imn)
+        new_im.save(im_path)
+
+
+if __name__ == '__main__':
+    num_map = {255: 1, 0: 0}
+    im_path = './masks/train'
+    suffix = 'png'
+    map_func(num_map, im_path, suffix)
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/tools/prepare_dataset.py:26]: This script seems doesn't work, could you double-check it?

File: projects/medical/2d_image/dermoscopy/ddti/README.md (hunk 1 above)
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/README.md:58]: After step 2, the content structure is like this:
```shell
data/DDTI
├── 1_or_data
│   ├── category.csv
│   ├── image
│   ├── mask
│   └── readme.txt
├── 2_preprocessed_data
│   ├── readme.txt
│   ├── stage1
│   ├── stage2
│   └── train.csv
└── readme.txt
```
It seems that can not directly use the `split_seg_dataset.py` script.

File: projects/medical/2d_image/dermoscopy/ddti/tools/prepare_dataset.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        for num in num_map:
+            imn[imn == num] = num_map[num]
+        new_im = Image.fromarray(imn)
+        new_im.save(im_path)
+
+
+if __name__ == '__main__':
+    num_map = {255: 1, 0: 0}
+    im_path = './masks/train'
+    suffix = 'png'
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/tools/prepare_dataset.py]: Can't find the folder at `data/DDTI/mask`.

Comment (xiexinch) [reply projects/medical/2d_image/dermoscopy/ddti/tools/prepare_dataset.py]: Should it modify like the following?
```python

if __name__ == '__main__':
    num_map = {255: 1, 0: 0}
    suffix = 'PNG'
    map_func(num_map, 'data/DDTI//1_or_data/mask', suffix)
```

File: projects/medical/2d_image/dermoscopy/ddti/README.md (hunk 1 above)
Comment (Provable0816) [reply projects/medical/2d_image/dermoscopy/ddti/README.md:58]: There is something wrong with our current dataset saved in ssd. I will check it later.

File: projects/medical/2d_image/dermoscopy/ddti/datasets/ddti_dataset.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+from mmseg.datasets import BaseSegDataset
+from mmseg.registry import DATASETS
+
+
+@DATASETS.register_module()
+class ddtiDataset(BaseSegDataset):
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/datasets/ddti_dataset.py:6]:
```suggestion
class DDTIDataset(BaseSegDataset):
```

File: projects/medical/2d_image/dermoscopy/ddti/configs/ddti_512x512.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+dataset_type = 'ddtiDataset'
```
Comment (xiexinch) [inline projects/medical/2d_image/dermoscopy/ddti/configs/ddti_512x512.py:1]:
```suggestion
dataset_type = 'DDTIDataset'
```

**Pull request closed without merging**